lib/streamDAO.js

```diff
@@ -74,12 +67,31 @@ function convertDatesToLegacy(entry) {
   }
 }
 
+// as part of migration process we attach the platform style
+function addDeduplicator(datum) {
+  datum._deduplicator = {
+    name: 'org.tidepool.deduplicator.device.deactivate.hash',
+    version: '1.1.0',
+    hash: misc.generateHash([
+      datum._userId,
+      datum.deviceId,
+      datum.time,
+      datum.type,
```

Comment:
Since this is using only the 4 "base" identity fields, won't this give different results from what platform currently does? As most datum types append additional fields in addition to these.

Reply:
In other words should this check the datum is an upload only?

Reply:
💡 ... ahhh i didn't realise that! Yes it would, make a difference.

Reply:
A few examples

https://github.com/tidepool-org/platform/blob/master/data/types/basal/basal.go#L63
https://github.com/tidepool-org/platform/blob/master/data/types/bolus/bolus.go#L80
https://github.com/tidepool-org/platform/blob/master/data/types/device/device.go#L58

Reply:
I believe I have corrected that now - thanks @lostlevels

Reply:
Unfortunately it will be more tedious than using the existing jellyfish idFields as is. As you can see jellyfish and platform use either different fields, or different order of fields for many types.

https://github.com/tidepool-org/platform/blob/master/data/types/basal/basal.go#L63
https://github.com/tidepool-org/jellyfish/blob/master/lib/schema/basal.js#L26

Reply:
thanks again @lostlevels. I have gone through for each type and add a separate `registerFieldsForDuplicator(` function based on what is in the platform code for that type